src/core/include/openvino/core/type/float8_e4m3.hpp

```diff
@@ -155,3 +155,66 @@ float8_e4m3 float8_e4m3::operator/=(const T& other) {
 #    pragma warning(pop)
 #endif
 }  // namespace ov
+
+namespace std {
+// bool OPENVINO_API isnan(ov::float8_e4m3 x); // TODO
```

Comment:
I think it is not required, because float8 should be implicitly cast to float when used in `std::isnan`.
Check and add test to confirm it, please.

Reply:
Removed, added corresponding tests for `std::isnan`